Composer/packages/client/src/components/CreationFlow/CreationFlow.tsx

```diff
@@ -113,7 +113,7 @@ const CreationFlow: React.FC<CreationFlowProps> = () => {
   };
 
   const handleCreateNew = async (formData, templateId: string, qnaKbUrls?: string[]) => {
-    createProject(
+    await createProject(
```

Comment:
@srinaath I notice you removed this 'await' when you submitted a PR a few days ago. My code relies on this call being awaited. What was the reason you removed the 'await' previously? I can refactor my code to account for this call not being awaited but it would be a rather large refactor.

As I see it, this should not be an issue if it is awaited. the createProject function handles triggering UI loading state, then it does all the work to create the project and then triggers UI state changes so the user is in the editing view.

In a non VA create flow, If it as awaited it will do all these, then return to this function and end execution, I could see it being slightly more optimal to not await but it creates an issue for my implementation. My code calls await handleCreateNew which calls createProject (assuming it is awaited) and then calls code that alters the bot files that it assumes are now created. If we dont await this call then my code runs bot asset altering code prior to the bot assets being stored in memory. 

Options
1. Revert this call to be awaited as it was prior to your change
2. Conditionally await depending on template

Reply:
So the way all the dispatcher functions need to work is that they should not be awaited inside the component. The way we built it is that it would be a "fire and forget" approach. We call the dispatcher, it does some updates to the state. We need to let the component subscribe to the change and do something in the useEffect of the change.

Reply:
That makes a lot of sense, many thanks for the tip and explanation. I will make the change now :)